docs/examples/onchain-nft.md

```diff
@@ -0,0 +1,361 @@
+---
+sidebar_label: OnChain NFT
+sidebar_position: 14
+---
+
+# Gear Non-Fungible Token
+
+### Introduction
+Non-fungible tokens (NFTs) are unique cryptographic tokens on a blockchain that are used to prove an ownership of a digital asset, such as digital art or gaming assets. The difference from fungible tokens is that the fungible tokens store a value, while non-fungible tokens store a cryptographic certificate.
+Under the hood, a non-fungible token consists of a unique token identifier, or token ID, which is mapped to an owner identifier and stored inside a NFT smart contract.<center> <em><strong>token_id</strong></em> → <em><strong>address</strong></em> </center>
+
+When the owner of a given token ID wishes to transfer it to another user, it is easy to verify ownership and reassign the token to a new owner.
+
+Mostly NFTs images (or other base resourse) is stored somewhere else (e.g. ipfs) and only the metadata is stored in the contract. Metadata consists of a name, an ID and links to the external resources, where the images is actually stored.
+But there is another approached introduced here. Sometimes you can store NFTs directly on chain without any external storage. This approach helps you not to lose your nft if there is a problem with the external storage.
+
+### Approach
+To successfully implement this approach several things are needed. Firstly, when initializing a collection, one should provide all the possible images of all the layers for a collection. Secondly, when minting alongside with a small metadata, one should provide a combination of layers used for a specific NFT. This approach seems quite costly when initializing, but is relatively cheap when it comes to minting.
+
+### Default non-fungible-token implementation
```

Comment:
Show the specific of on-chain NFT, keep only necessary info that needs to understand the difference of common NFT implementation from on-chain NFT.